packages/graphql/src/schema/make-augmented-schema.ts

```diff
@@ -763,9 +783,30 @@ function makeAugmentedSchema(
         generatedResolvers[union.name.value] = { __resolveType: (root) => root.__resolveType };
     });
 
+    const seen = {};
+    let parsedDoc = parse(generatedTypeDefs);
+    parsedDoc = {
```

Comment:
Trying to get my head around this but can't quite figure out what it's doing... Any chance of a quick explanation?

Reply:
It filters out the seen definitions in the document, without would result in an error `Cannot have two types with the name of Int`.